### PR TITLE
feat: allow setting marker type for MarkerGroups

### PR DESCRIPTION
**Pull request by akoreman**
*Issue #, if available:* NA

*Description of changes:* When using `MarkerGroup`s, currently, all markers default to being of the `text` type in which only the text itself is highlighted within a range. Users of `MarkerGroup`s might want to have different types of markers. 

This change adds an `options` parameter to allow setting marker type to `fullLine` or `line`. In `fullLine`, all the rows within the range will be fully highlighted, for `line` we highlight the same as default but when newlines are part of the range we fill up the highlight from the last character within the line to the right side of the editor.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

Pull Request Checklist:
* [x] No backwards incompatible changes were made to Ace's public interface which is defined through the main typings file (`ace.d.ts`) and its references:
    * https://github.com/ajaxorg/ace/blob/master/ace.d.ts 
    * https://github.com/ajaxorg/ace/blob/master/ace-modes.d.ts 
    * https://github.com/ajaxorg/ace/blob/master/ace-extensions.d.ts

### CODE REVIEW

File: src/marker_group.js [hunk 1]
```diff
@@ -59,12 +64,12 @@ class MarkerGroup {
...
      * @param {Marker} markerLayer
      * @param {EditSession} session
-     * @param {{ firstRow: any; lastRow: any; }} config
+     * @param {LayerConfig} config
      */
     update(html, markerLayer, session, config) {
         if (!this.markers || !this.markers.length)
             return;
-        var visibleRangeStartRow = config.firstRow, visibleRangeEndRow = config.lastRow;
+        var visibleRangeStartRow = config.firstRow, visibleRangeEndRow = 100;
```
Comment (whazor) [inline src/marker_group.js:72]: should this be changed?

Comment (akoreman) [reply src/marker_group.js:72]: ah whoops, that was just a change I made while playing around thanks for catching that 😅

File: src/marker_group.js [hunk 2]
```diff
@@ -15,8 +16,12 @@ Potential improvements:
 class MarkerGroup {
     /**
      * @param {EditSession} session
+     * @param {{markerType: "fullLine" | "line" | undefined}} [options]
```
Comment (whazor) [inline src/marker_group.js:19]: could you describe here what each argument does?

File: src/marker_group.js (hunk 1 above)
Comment (akoreman) [reply src/marker_group.js]: Removed that change

File: src/marker_group.js (hunk 2 above)
Comment (akoreman) [reply src/marker_group.js]: added a JSDoc comment

**Review by whazor — APPROVED** [review]

**Pull request merged**